src/test/java/org/wso2/carbon/connector/pcml/test/unit/AS400TraceUnitTest.java

```diff
@@ -0,0 +1,121 @@
+import org.testng.annotations.Test;
+import org.wso2.carbon.connector.core.ConnectException;
+import org.wso2.carbon.connector.pcml.AS400Constants;
+import org.wso2.carbon.connector.pcml.AS400Trace;
+
+import java.io.File;
+import java.io.StringReader;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Stack;
+import javax.xml.stream.XMLInputFactory;
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.XMLStreamReader;
+
+public class AS400TraceUnitTest {
+    private MessageContext messageContext;
+    private TemplateContext templateContext;
+    private Stack functionStack;
+    private AS400Trace as400Trace;
+    private static final String CARBON_HOME = "carbon.home";
+
+    @BeforeMethod
+    public void setUp() throws Exception {
+        as400Trace = new AS400Trace();
+        messageContext = createMessageContext();
+        templateContext = new TemplateContext("AS400Init", null);
+        functionStack = new Stack();
+    }
+
+    public static void setCarbonHome() {
+        Path carbonHome = Paths.get("");
+        carbonHome = Paths.get(carbonHome.toString());
+        System.setProperty(CARBON_HOME, carbonHome.toAbsolutePath().toString());
+        Path create = Paths.get(carbonHome.toString(), "repository", "logs");
+        File file = new File(create.toAbsolutePath().toString());
+        if (!file.exists()) {
+            file.mkdirs();
+        }
+    }
+
+    @Test(description = "Test AS400Trace Connect")
+    public void testAs400TraceConnecct() throws ConnectException {
+        setCarbonHome();
+        setTraceProperties();
+        functionStack.push(templateContext);
+        messageContext.setProperty("_SYNAPSE_FUNCTION_STACK", functionStack);
+        as400Trace.connect(messageContext);
+        Assert.assertEquals(Trace.getFileName(),
+                "/home/jananithangavel/Desktop/new_unitTest/esb-connector-pcml/target/repository/logs/pcml-connector-logs.log");
+        Assert.assertFalse(Trace.isTraceAllOn());
+    }
+
+    public void setTraceProperties() {
+        templateContext.getMappedValues().put(AS400Constants.AS400_TRACE_CONVERSION, "true");
+        templateContext.getMappedValues().put(AS400Constants.AS400_TRACE_DATASTREAM, "true");
+        templateContext.getMappedValues().put(AS400Constants.AS400_TRACE_DIAGNOSTICS, "false");
+        templateContext.getMappedValues().put(AS400Constants.AS400_TRACE_ERROR, "true");
+        templateContext.getMappedValues().put(AS400Constants.AS400_TRACE_INFORMATION, "true");
+        templateContext.getMappedValues().put(AS400Constants.AS400_TRACE_PCML, "true");
+        templateContext.getMappedValues().put(AS400Constants.AS400_TRACE_WARNING, "true");
+        templateContext.getMappedValues().put(AS400Constants.AS400_TRACE_PROXY, "false");
+        templateContext.getMappedValues().put(AS400Constants.AS400_TRACE_ALL, "false");
+    }
+
+    public MessageContext createMessageContext() throws AxisFault {
```

Comment:
Javadoc comments missing. Also duplicate method.

Please move to a separate class and reuse it in tests.